rubric_data/exceptions.py

```diff
@@ -0,0 +1,19 @@
+class Error(Exception):
+    """Base class for other exceptions"""
+    
+    def __init__(self, message="No message defined."):
+        self.message = message
+
+class StudentsSectionsError(Error):
+    """Error in extracting the Students and Sections from the API response"""
+    pass
+
+
+class RubricAssignmentsError(Error):
+    """Error in extracting the RubricAssignments from the API response"""
+    pass
+
+
+class DatapointsError(Error):
```

Comment:
Looking at how the exceptions are being used, I wonder if there would be any advantage to elevating the category of error over the model associated with that error. For example:
- `EmptyDataError` could be thrown if the input data is empty
- `DataProcessingError` could be thrown if there's a problem with the data processing (e.g. handle `KeyError`)

This would allow you to more easily differentiate those classes of errors and potentially return different responses to the frontend if needed. OTOH the current exceptions allow you to quickly identify the model that had the problem.

Reply:
I see what you mean, and I think your approach is sound, especially if the domain model grows in complexity. As it is though, I am passing a message to the error and logging that message with what, I think, is all the information we need at the moment.